### ISSUE TITLE
Build using Carthage?

### ISSUE THREAD

**Issue by davenquinn** [GEOSwift/GEOSwift#39]
Hi,

This looks like fantastic work! I am trying to integrate into an app where I am using Carthage as my framework manager — it's a little more lightweight than CocoaPods. I am wondering if it is possible to configure this for Carthage...I tried installing it and there was some cryptic (to me) error which I have pasted below. Perhaps this just means that there needs to be a Cartfile in the project directory specifying dependencies (MapBox iOS SDK?). Might be...anyway, insight from other interested parties would be appreciated.

Thanks,
Daven

```
*** Building scheme "GEOSwift" in GEOSwift.xcworkspace
Build Failed
	Task failed with exit code 65:
	/usr/bin/xcrun xcodebuild -workspace "/Users/Daven/Development/Mapping App/Map-Digitizer/Carthage/Checkouts/GEOSwift/GEOSwift.xcworkspace" -scheme GEOSwift -configuration Release -derivedDataPath /Users/Daven/Library/Caches/org.carthage.CarthageKit/DerivedData/GEOSwift/f5a631fe99cdac1fd72fd680763b8cf952a9aed6 -sdk iphoneos ONLY_ACTIVE_ARCH=NO BITCODE_GENERATION_MODE=bitcode CODE_SIGNING_REQUIRED=NO CODE_SIGN_IDENTITY= CARTHAGE=YES clean build

This usually indicates that project itself failed to compile. Please check the xcodebuild log for more details: /var/folders/md/lvb3zv4n5652ffl6t44397t80000gn/T/carthage-xcodebuild.l8D67B.log
```

**Comment by vfn** [comment]
Hi Daven,

Can you please paste the log mentioned in the last line of you log above? `This
usually indicates that project itself failed to compile. Please check the
xcodebuild log for more details: /var/folders/md/
lvb3zv4n5652ffl6t44397t80000gn/T/carthage-xcodebuild.l8D67B.log`

If you can't find that log, try build the project again using Carthage, and
then look for the new log file.

Cheers,
V

**Comment by davenquinn** [comment]
Oh sorry, that's a pretty obvious step for me to take! My bad. It was kinda large so I attached below...

[xcode-build.txt](https://github.com/andreacremaschi/GEOSwift/files/895448/xcode-build.txt)

**Comment by kylebrowning** [comment]
Getting this as well.

**Comment by kylebrowning** [comment]
```/Users/kylebrowning/work/ios/leagueapp/Carthage/Checkouts/GEOSwift/GEOSwift.playground' in workspace '/Users/kylebrowning/work/ios/leagueapp/Carthage/Checkouts/GEOSwift/GEOSwift.xcworkspace```

**Comment by vfn** [comment]
@samritchie Would you be able to create a pull request with what you are suggesting?

Thanks!

**Comment by samritchie** [comment]
Can I verify if the playground should be working? I only get ```Playground execution failed:

error: Couldn't lookup symbols:
  __T08GEOSwift8GeometryC7crossesSbACF
  __T08GEOSwift8WaypointCMa
  __T08GEOSwift8GeometryC8disjointSbACF
  __T08GEOSwift12MultiPolygonCMa
  __T08GEOSwift8GeometryC6relateSbAC_SS7patterntF
  __T08GEOSwift8GeometryC12intersectionACSgACF
  __T08GEOSwift8GeometryC6equalsSbACF
  __T08GEOSwift8GeometryCACSgSS3WKT_tcfC
  __T08GEOSwift8GeometryCMa
  __T08GEOSwift8GeometryC6bufferACSgSd5width_tF
  __T08GEOSwift8GeometryC8containsSbACF
  __T0Sa8GEOSwiftAA7FeatureCRszlE11fromGeoJSONSayACGSg10Foundation3URLVKFZ
  __T08GEOSwift8GeometryC5unionACSgACF
  __T08GEOSwift7PolygonCMa
  __T08GEOSwift8GeometryC7touchesSbACF
  __T08GEOSwift8GeometryC10differenceACSgACF
  __T08GEOSwift8GeometryC8boundaryACSgyF
  __T08GEOSwift8GeometryC8overlapsSbACF
  __T08GEOSwift8GeometryC14pointOnSurfaceAA8WaypointCSgyF
  __T08GEOSwift8GeometryC10convexHullAA7PolygonCSgyF
  __T08GEOSwift8GeometryC6withinSbACF
  __T08GEOSwift7FeatureCMa
  __T08GEOSwift8GeometryC8centroidAA8WaypointCSgyF
  __T08GEOSwift8GeometryC10intersectsSbACF
  __T08GEOSwift8GeometryC8mapShapeSo7MKShapeCyF
  __T08GEOSwift8GeometryC8envelopeACSgyF```

**Comment by vfn** [comment]
@samritchie it was working until Xcode 9.2, that was the last time I checked. I haven't tried it on Xcode 9.3

**Comment by mmarkov-appolica** [comment]
Is there a working solution for the carthage?

**Comment by madhavajay** [comment]
https://github.com/GEOSwift/GEOSwift/pull/145

**Comment by madhavajay** [comment]
https://github.com/GEOSwift/GEOSwift/pull/149

**Comment by macdrevx** [comment]
#153

**Comment by madhavajay** [comment]
@macdrevx and I have solved the issue. For anyone who wants to know what we did. I created an xcode project for the https://github.com/GEOSwift/geos dependency which allows a nested Carthage dependency in GEOSwift.

There was a hacky way to then get Carthage to play ball, by having a pre-script which copied the geos.framework into the expected filename for the existing Cocoapods Workspace setup.
However @macdrevx opted instead to replace Cocoapods with Carthage as the primary Dependency manager for the GEOSwift project, thus allowing Cocoapods to just be a spec for external use.

The final missing piece of the puzzle was getting the Playgrounds to work. Turns out that the playgrounds are able to find frameworks in the current Build > Product > Arch directory $BUILT_PRODUCTS_DIR so I simply copy the geos.framework from the Carthage folder into this local folder during build so that the Playground sees it.

If anyone knows a better solution I am all ears. I tried the obvious solution of nesting the Playground in the project since the Framework Search Paths are defined there and work for the framework build, but no luck.

Anyway, looks like GEOSwift is now: Cocoapods, Carthage and SPM compatible which is great news!
